bee-tangle/src/lib.rs

```diff
@@ -7,15 +7,31 @@ use tokio::sync::RwLock;
 
 use std::{collections::HashMap, sync::Arc};
 
-struct MessageData {
-    message: Arc<Message>,
+/// Data structure used by the [`Tangle`] to store a [`Message`] with its associated [`MessageMetadata`].
+pub struct MessageData {
+    message: Message,
     metadata: MessageMetadata,
 }
 
+impl MessageData {
+    /// Retrieves both the [`Message`] and the [`MessageMetadata`] from the [`MessageData`].
+    pub fn get(&self) -> (&Message, &MessageMetadata) {
```

Comment:
I don't think this one is necessary tbh, it's a weird API when you have the data IMO

Reply:
removed it